README.md

```diff
@@ -1,8 +1,336 @@
+        nodes {
+          title
+        }
+        pageInfo {
+          hasNextPage
+          endCursor
+        }
+      }
+    }
+  }`
+);
+
+for await (const response of pageIterator) {
+  const issues = response.repository.issues;
+  console.log(`${issues.length} issues found.`);
+}
+```
+
+### Variables
+
+Per default, the plugin creates a query-statement (like `query paginate(cursor1: String!)`) containing the cursor variable(s) you provided.
+
+To pass your own variables, you can create the query-statement yourself and pass the variables as a second parameter to the `paginateGraphql` or `iterator`-function, just like you do with the [octokit/graphql.js](https://github.com/octokit/graphql.js/#variables) plugin.
+
+```js
+await octokit.paginateGraphql(
+  (cursor) => {
+    const cursorVariable = cursor.create();
+    return `
+      query paginate(${cursorVariable}: String, $organization: String!) {
+        repository(owner: $organization, name: "rest.js") {
+          issues(first: 10, after: ${cursorVariable}) {
+            nodes {
+              title
+            }
+            pageInfo {
+              hasNextPage
+              endCursor
+            }
+          }
+        }
+      }
+    `;
+  },
+  {
+    organization: "octokit",
+  }
+);
+```
+
+> **Note**
+> You will have to make sure to insert the created cursor-variable both in the query statement as well as in the paginated resource.
+
+### Initial cursor values
+
+To pass initial cursor values, you can create a named cursor by passing a string to the cursor-creator like `cursor.create("namedCursor")` and then use the name as property-key in the variable-object:
+
+```js
+await octokit.paginateGraphql(
+  (cursor) => {
+    const cursorVariable = cursor.create("namedCursor");
+    return `
+      query paginate(${cursorVariable}: String, $organization: String!) {
+        repository(owner: $organization, name: "rest.js") {
+          issues(first: 10, after: ${cursorVariable}) {
+            nodes {
+              title
+            }
+            pageInfo {
+              hasNextPage
+              endCursor
+            }
+          }
+        }
+      }
+    `;
+  },
+  {
+    namedCursor: "initialValue",
+    organization: "octokit",
+  }
+);
+```
+
+### Pagination Direction
+
+You can control the pagination direction by the structure of the provided `pageInfo`.
+
+For a forward pagination, use:
+
+```gql
+pageInfo {
+  hasNextPage
+  endCursor
+}
+```
+
+For a backwards pagination, use:
+
+```gql
+pageInfo {
+  hasNextPage
+  endCursor
+}
+```
+
+If you provide all 4 properties in a `pageInfo`, the plugin will default to forward pagination.
+
+### Parallel pagination
+
+You can do a parallel pagination by creating several cursors for different resources:
+
+```js
+const { repository } = await octokit.paginateGraphql((cursor) => {
+  const topicsCursor = cursor.create("topics");
+  const issuesCursor = cursor.create("issues");
```

Comment:
given that the order matters, shouldn't we recommend to set the cursors inline instead of using variables?

Reply:
Won't be an issue without parallel pagination anymore. 🙂